ERCS/erc-7798.md

```diff
@@ -0,0 +1,200 @@
+  rpcProxySubmissionParams?: {
+    submissionUrl?: string; 
+  }
+}
+```
+
+**Contract Call**
+
+```ts
+{
+  payloadType: 'contractCall';
+  chainId: string;
+  approveTxs?: {
+    data: string; // this represents the approval call data
+    toAddress: string; // the address to submit the approve transaction to 
+  }[],
+  paymentTx: {
+    data: string; // call data of the transaction, only needed for 
+    toAddress: string; // the address to submit the payment transaction to
+    value: string; // the value of the blockchain's native token to transfer over  
+  }, 
+  rpcProxySubmissionParams?: {
+    submissionUrl?: string; // optional endpoint to submit the tx hash to
+  },
+  dappUrl?: string;
+  dappName?: string;
+}
+```
+
+**[EIP-712](https://eips.ethereum.org/EIPS/eip-712) Message (offline signature)**
+
+```ts
+{
+  payloadType: 'eip712';
+  chainId: string,
+  rpcProxySubmissionParams: {
+    submissionUrl: string; // endpoint to submit tx message + signature to 
+    typedData: {
+      types: {
+        // any types can go here in accordance with EIP-712 and eth_signtypeddata_v4, it's recommended that every type needed for
+        // generating the right transactions on the RPC proxy is here
+      },
+      primaryType?: string;
+      domain: {
+        // any domain fields go here, usually need name, version, chainId and verifyingContract
+      },
+      message: {
+        // any fields can go here, it's recommended that every field needed for
+        // generating the right transactions on the RPC proxy is here
+      }
+    },
+  },
+  dappUrl?: string;
+  dappName?: string;
+  additionalPayload?: {
+    // ... any fields can go here, this is additional payload the RPC proxy may need but does not have to bee signed in the message
+  }
+}
+```
+Upon submitting the transaction for the [ERC-681](erc-681.md) and contract call cases, if `rpcProxySubmissionParams` is present, then the transaction hash can be optionally submitted via a POST request to the submission URL. The body of the POST would have the following structure:
+```ts
+{
+  txHash: string; 
+}
+```
+
+## Rationale
+
+The reason for having a relayer URI to pass data is due to limitations in the size of the data that can be transmitted wirelessly between devices directly. For example, QR codes have a maximum character size of 7,089 characters currently, but a user can attempt to buy an indefinite number of items in a single checkout transaction. By having a relayer URI, we can theoretically pass as much calldata as we desire to the customer’s wallet, as long as they fetch from the URI.
+
+[ERC-681](erc-681.md) URIs were also permitted to enable possible direct peer to peer payments. It’s more likely that they’ll be occurring between two end user wallets rather than between a customer and a merchant, but it would be good to include in the event the dapp only requires a simple send transaction to execute the checkout.
+
+## Security Considerations
```

Comment:
NFC isn’t end-to-end encrypted, so I think there’s a potential vulnerability to a relay attack. An adversary intercepts and forwards communication between a user and a merchant, basically extending the NFC range. To mitigate this, wallets should have security precautions like transaction simulation, a trusted merchants list, or spending limits.